src/serlo-editor/plugins/text/hooks/use-slate-render-handlers.tsx

```diff
@@ -80,6 +80,31 @@ export const useSlateRenderHandlers = ({
           </MathElement>
         )
       }
+      if (element.type === 'gap') {
+        // TODO: Render <GapRenderer> here instead
+        // <GapRenderer> needs to ...
+        // - always show input fields
+        // - accept callback onChange that is passed to input element to update correctAnswer
+        return (
+          <span {...attributes} contentEditable={false}>
+            <input
+              value={element.correctAnswer}
+              size={element.correctAnswer.length}
```

Comment:
What I noticed while using Duolingo, is that they will always add a bit of size, to not make it too obvious what the solution is. So, maybe like `size={element.correctAnswer.length + 2}` or something.

Reply:
Good idea! However, this input is only visible in the editor. The input in the no-edit view will size itself based on the users input (not based on the correct answer), so the size will not give anything away there. However, for drag and drop this might become important later because one could find the correct solution by comparing the size of the gap to the size of the draggable element.